src/proposer.ts

```diff
@@ -641,8 +624,7 @@ async function handleIntention(
 	 */
 	for (const input of validatedIntention.inputs) {
 		const tokenAddress = input.asset
-		const tokenDecimals = await getTokenDecimals(tokenAddress)
-		const requiredAmount = parseUnits(input.amount, Number(tokenDecimals))
+		const requiredAmount = safeBigInt(input.amount)
```

Comment:
Do we need to check decimals somewhere else since the tokens may not all be in `wei` (18 decimals)?